src/operations.c

```diff
@@ -25,14 +25,69 @@
 #include "baton.h"
 #include "operations.h"
 
+// Mutex protecting the connection and the run_timeout_thread flag
+pthread_mutex_t conn_mutex = PTHREAD_MUTEX_INITIALIZER;
+// The connection used by iterate_json and connection_timeout
+rcComm_t *connection;
+// While true, the client will continue to run the timeout thread
+int run_timeout_thread = 1;
+// Condition variable to exit the timeout thread when work is complete
+pthread_cond_t watchdog_cond = PTHREAD_COND_INITIALIZER;
+
+// Refresh the connection every timeout seconds
+void *connection_timeout(void *timeout) {
+    int tsec = *((int *) timeout);
+
+    struct timespec abs_timeout;
+
+    while (1) {
+        pthread_mutex_lock(&conn_mutex);
+        if (!run_timeout_thread) {
+            pthread_mutex_unlock(&conn_mutex);
+            break;
+        }
+
+        clock_gettime(CLOCK_REALTIME, &abs_timeout); // CLOCK_MONOTONIC instead?
```

Comment:
`CLOCK_REALTIME` is the default one used by `pthread_cond_timedwait()`, so this needs to match.  On some systems [you can switch to `CLOCK_MONOTONIC`](https://linux.die.net/man/3/pthread_condattr_setclock) but support for the functions needed is patchy and you may not have them.

In practice using `CLOCK_REALTIME` shouldn't cause too much trouble as long as no-one makes a huge backwards clock adjustment while this is running.  If they do then you'll just get a rather longer timeout than you expected.

Reply:
Sounds like it's more trouble than it's worth 👍